rclc/src/rclc/init.c

```diff
@@ -63,7 +64,8 @@ rclc_support_init_with_options(
     allocator, "allocator is a null pointer", return RCL_RET_INVALID_ARGUMENT);
   rcl_ret_t rc = RCL_RET_OK;
 
-  memcpy(&support->init_options, init_options, sizeof(rcl_init_options_t));
+  support->init_options = rcl_get_zero_initialized_init_options();
+  rcl_init_options_copy(init_options, &support->init_options);
```

Comment:
This uses another memory node before its released here:
https://github.com/ros2/rclc/blob/4cd7e692eaf4d6229f215b5fdd89d4e3924e2a0a/rclc/src/rclc/init.c#L45-L46

Also, the user may call this function with their own initialized options, so its their responsibility to call the cleanup method.